src/backend/distributed/commands/function.c

```diff
@@ -1443,10 +1443,18 @@ DefineAggregateStmtObjectAddress(Node *node, bool missing_ok)
 	ObjectWithArgs *objectWithArgs = makeNode(ObjectWithArgs);
 	objectWithArgs->objname = stmt->defnames;
 
-	FunctionParameter *funcParam = NULL;
-	foreach_ptr(funcParam, linitial(stmt->args))
+	if (stmt->args != NIL)
+	{
+		FunctionParameter *funcParam = NULL;
+		foreach_ptr(funcParam, linitial(stmt->args))
+		{
+			objectWithArgs->objargs = lappend(objectWithArgs->objargs,
+											  funcParam->argType);
+		}
+	}
+	else
 	{
-		objectWithArgs->objargs = lappend(objectWithArgs->objargs, funcParam->argType);
+		objectWithArgs->objargs = list_make1(makeTypeName("anyelement"));
```

Comment:
Why do we need to have the else part? Can't we have objectWithArgs with empty objargs?

Reply:
Seems a bit odd to me as well, but as far as I understand, the reason is:

```sql
CREATE AGGREGATE first (
  sfunc    = first_agg,
  basetype = anyelement,
   = any  stype    = anyelement,
  combinefunc = first_agg
  );
select pronargs from pg_proc where proname='first';
┌──────────┐
│ pronargs │
├──────────┤
│        1 │
└──────────┘
(1 row)
```

Seems pg stores aggregates with no args as with one arg which is "anyelement".

Without the else part, the pg function `LookupFuncWithArgs` doesn't find the aggregate, i.e it requires an input with one arg named "anyelement".

Reply:
Can you add such a test to regression tests? I know we can create aggregates without a parameter using the old syntax just like you used here, but I don't how we can create aggregates without a parameter using the new syntax. Can you please check and add a test for it as well?